src/clojupyter/misc/kind.clj

```diff
@@ -0,0 +1,464 @@
+(ns clojupyter.misc.kind
+  (:require
+   [cheshire.core :as cheshire]
+   [clojupyter.display :as display]
+   [clojure.data.codec.base64 :as b64]
+   [clojure.java.io :as io]
+   [scicloj.kindly-render.note.to-hiccup :as to-hiccup]
+   [scicloj.kindly-render.shared.walk :as walk]
+   [scicloj.kindly-advice.v1.api :as kindly-advice]
+   [clojure.string :as str])
+  (:import
+   [javax.imageio ImageIO]))
+
+(defn require-js
+  "Generates a Hiccup representation of a `<script>` tag that dynamically loads a JavaScript library from  
+   a given URL and ensures that a specific JavaScript object provided by the library is available before  
+   executing a rendering command. This is used to include external JavaScript libraries in a Jupyter notebook environment.  
+  
+   **Parameters:**  
+  
+   - `url` (String): The URL of the JavaScript library to load.  
+   - `js-object` (String): The name of the JavaScript object that the library defines (e.g., `'Plotly'`, `'Highcharts'`).  
+   - `render-cmd` (String): The JavaScript code to execute after the library has been loaded and the object is available.  
+  
+   **Returns:**  
+  
+   - A Hiccup vector representing a `<script>` tag containing JavaScript code that loads the library and executes `render-cmd` once the library is loaded."
+  [url js-object render-cmd]
+  [:script
+   (format
+    "  
+    loadScript_%s = src => new Promise(resolve => {  
+    script_%s = document.createElement('script');  
+    script_%s.src = src;  
+    script_%s.addEventListener('load', resolve);  
+    document.head.appendChild(script_%s);  
+  });  
+  if (typeof %s === 'undefined') {    
+     promise_%s=loadScript_%s('%s')  
+       
+     Promise.all([promise_%s]).then(() => {  
+       console.log('%s loaded');  
+        })  
+       
+     };  
+ %s    
+  
+ "
+    js-object js-object js-object
+    js-object js-object js-object js-object
+    js-object url js-object js-object
+    render-cmd)])
+
+(defn require-cytoscape
+  "Creates a Hiccup representation to load the Cytoscape.js library and execute a rendering command after it has been loaded.  
+  
+   **Parameters:**  
+  
+   - `render-cmd` (String): The JavaScript command to execute after Cytoscape.js has been loaded.  
+  
+   **Returns:**  
+  
+   - A Hiccup vector that includes a `<script>` tag loading Cytoscape.js and executing the provided rendering command."
+  [render-cmd]
+  (require-js "https://cdnjs.cloudflare.com/ajax/libs/cytoscape/3.30.4/cytoscape.min.js"
```

Comment:
Non-blocking but it would be nice to have the script versions determined dynamically rather than hardcoded into the kernel. I'll create an issue

Reply:
I agree. Maybe this code will be "moved" to "kindly-render"